app/v1/asset/keeper.go

```diff
@@ -62,8 +62,10 @@ func (k Keeper) CreateGateway(ctx sdk.Context, msg MsgCreateGateway) (sdk.Tags,
 		Operators:  msg.Operators,
 	}
 
-	// save the gateway with the creation enabled
-	k.saveGateway(ctx, gateway, true)
+	// set the gateway and related keys
+	k.SetGateway(ctx, gateway)
+	k.SetMoniker(ctx, msg.Moniker, gatewayID)
+	k.SetOwnerGatewayID(ctx, msg.Owner, gatewayID)
```

Comment:
Does this mean that one owner can only create one gateway ?

Reply:
> Does this mean that one owner can only create one gateway ?

No. This key is used to iterate ids on an owner with prefix. Its format is "gateways:<owner>:<id>"